sdk/model/task_json_io.go

```diff
@@ -0,0 +1,157 @@
+	StartTime                        int64            `json:"startTime,omitempty"`
+	EndTime                          int64            `json:"endTime,omitempty"`
+	UpdateTime                       int64            `json:"updateTime,omitempty"`
+	StartDelayInSeconds              int32            `json:"startDelayInSeconds,omitempty"`
+	RetriedTaskId                    string           `json:"retriedTaskId,omitempty"`
+	Retried                          bool             `json:"retried,omitempty"`
+	Executed                         bool             `json:"executed,omitempty"`
+	CallbackFromWorker               bool             `json:"callbackFromWorker,omitempty"`
+	ResponseTimeoutSeconds           int64            `json:"responseTimeoutSeconds,omitempty"`
+	WorkflowInstanceId               string           `json:"workflowInstanceId,omitempty"`
+	WorkflowType                     string           `json:"workflowType,omitempty"`
+	TaskId                           string           `json:"taskId,omitempty"`
+	ReasonForIncompletion            string           `json:"reasonForIncompletion,omitempty"`
+	CallbackAfterSeconds             int64            `json:"callbackAfterSeconds,omitempty"`
+	WorkerId                         string           `json:"workerId,omitempty"`
+	OutputData                       json.RawMessage  `json:"outputData,omitempty"`
+	WorkflowTask                     *WorkflowTask    `json:"workflowTask,omitempty"`
+	Domain                           string           `json:"domain,omitempty"`
+	RateLimitPerFrequency            int32            `json:"rateLimitPerFrequency,omitempty"`
+	RateLimitFrequencyInSeconds      int32            `json:"rateLimitFrequencyInSeconds,omitempty"`
+	ExternalInputPayloadStoragePath  string           `json:"externalInputPayloadStoragePath,omitempty"`
+	ExternalOutputPayloadStoragePath string           `json:"externalOutputPayloadStoragePath,omitempty"`
+	WorkflowPriority                 int32            `json:"workflowPriority,omitempty"`
+	ExecutionNameSpace               string           `json:"executionNameSpace,omitempty"`
+	IsolationGroupId                 string           `json:"isolationGroupId,omitempty"`
+	Iteration                        int32            `json:"iteration,omitempty"`
+	SubWorkflowId                    string           `json:"subWorkflowId,omitempty"`
+	SubworkflowChanged               bool             `json:"subworkflowChanged,omitempty"`
+	QueueWaitTime                    int64            `json:"queueWaitTime,omitempty"`
+	TaskDefinition                   *TaskDef         `json:"taskDefinition,omitempty"`
+	LoopOverTask                     bool             `json:"loopOverTask,omitempty"`
+}
+
+func (task *PolledTask) ToValue(newValue any) (interface{}, error) {
+
+	bytes, err := task.InputData.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(bytes, newValue)
+	if err != nil {
+		return nil, err
+	}
+	return newValue, nil
+}
```

Comment:
directly unmarshalling task.InputData into newValue
```suggestion
func (task *PolledTask) ToValue(newValue interface{}) (interface{}, error) {
    err := json.Unmarshal(task.InputData, newValue)
    if err != nil {
        return nil, err
    }
    return newValue, nil
}
```